cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -590,8 +590,8 @@ addLovelace a b = assumeBound $ CC.Common.addLovelace a b
 -----------------------------------------------------------------------------------------
 
 -- | Set of available tx outputs (for new transactions).
-availableFunds :: TVar.TVar AvailableFunds
-availableFunds = unsafePerformIO $ TVar.newTVarIO Set.empty
+availableFunds :: STM.STM (TVar.TVar AvailableFunds)
+availableFunds = TVar.newTVar Set.empty
```

Comment:
Now each time you call `availableFunds`, you'll create a new `TVar` :confused:

Reply:
Why? I just removed `unsafePerformIO`, so now `TVar` isn't fetched from `STM`-context, right?

Reply:
Maybe if you inline the definition of `availableFunds`, you'll see what I mean. Have you tried running the code?

Reply:
Not yet, now I just want to understand how to remove _all_ `unsafePerformIO`...

Reply:
@mrBliss Yes, it doesn't work..

Reply:
@denisshevchenko Suggestion: create the `TVar`s you need once, upon startup, and pass them to the functions that'll need to read/write to them. If you find yourself passing a lot of arguments to a bunch of functions, consider bundling them in a record and/or using `ReaderT`.

Disclaimer: I have not looked at your code, only the diff. I'm assuming you are using `TVar`s because you are sending things between multiple threads. If that's not the case, e.g., if you are using them as mutable variables, then `TVar`s are not the right solution.